pkg/internal/controller/controller.go

```diff
@@ -353,3 +381,46 @@ func (c *Controller) InjectFunc(f inject.Func) error {
 func (c *Controller) updateMetrics(reconcileTime time.Duration) {
 	ctrlmetrics.ReconcileTime.WithLabelValues(c.Name).Observe(reconcileTime.Seconds())
 }
+
+// KRef returns ObjectRef from name and namespace
+// Note: This is a copy of the func from klog. It has been copied to avoid
+// introducing a dependency to klog, while still implement logging according
+// to the Kubernetes structured logging KEP.
```

Comment:
This is based on the assumption that we don't want to introduce a dependency to klog.
I don't know if that's true.

Reply:
We definitely can